### PR TITLE
Plotting small geometries: A fix

### PR DESCRIPTION
**Pull request by nicolejkeeney**
This PR addresses the issue of plotting gridded data that's only one grid cell. Previously a bug had been raised by ```app.explore.warming_levels()```. Also, in ```app.view()```, data that was only one grid cell would output a weird lineplot with simulation or scenario on the axis. Now, data with only one grid cell is shown as a scatter plot. 

**To test**: Confirm that app.view() still works-- check a view geometries, station data, area average, etc. To make sure I didn't *introduce* any bugs with this bug fix. Confirm that ```app.explore.warming_levels()``` works for small geometries (i.e. a small county at 45km resolution)

### CODE REVIEW

**Comment by vicford** [comment]
In app.explore.warming, also using Alpine County 45 km getting.... two blue dots with no data??
Screen Shot 2023-05-10 at 10 09 04 AM

**Comment by nicolejkeeney** [comment]
> In app.explore.warming, also using Alpine County 45 km getting.... two blue dots with no data?? Screen Shot 2023-05-10 at 10 09 04 AM

Okay, I think I fixed this! :D

File: climakitae/view.py [hunk 1]
```diff
@@ -79,92 +79,77 @@ def _visualize(data, lat_lon=True, width=None, height=None, cmap=None):
...
                     height=height,
                 )
             else:
-                # Create map
-                _plot = data.hvplot.image(
-                    x="lon",
-                    y="lat",
+                # Make a scatter plot if it's just one grid cell
+                print(
+                    "Warning: your input data has 2 or less gridcells. Due to plotting limitations for small areas, a scatter plot will be generated."
```
Comment (vicford) [inline climakitae/view.py:136]: While I'm not a huge fan of the scatter plot, I do think this is an appropriate solution to do a scatter plot on too large resolutions. Could we potentially modify this warning to suggest trying a finer resolution (when reasonable)?

**Review by vicford — APPROVED** [review]
Data values now showing up in hover :)

**Comment by bethem** [comment]
in app.explore.warming(), when selecting Santa Cruz County @ 45 km (emphasis mine, since there are two main errors shown):

`Traceback (most recent call last):
  File "/home/jovyan/climakitae/data_loaders.py", line 425, in _get_data_one_var
    dset = dset.rio.clip(geometries=ds_region, crs=4326, drop=True)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_dataset.py", line 380, in clip
    self._obj[var]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_array.py", line 941, in clip
    raise NoDataInBounds(
**rioxarray.exceptions.NoDataInBounds: No data found in bounds. Data variable: t2**

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pyviz_comms/__init__.py", line 346, in _handle_msg
    self._on_msg(msg)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/viewable.py", line 291, in _on_msg
    patch.apply_to_document(doc, comm.id if comm else None)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in apply_to_document
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc
    return f()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in <lambda>
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/document.py", line 391, in apply_json_patch
    DocumentPatchedEvent.handle_json(self, event_json, references, setter)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 259, in handle_json
    handler(doc, event_json, references, setter)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 300, in _handle_json
    cb(event_json["msg_data"])
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 355, in trigger_json_event
    model._trigger_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 119, in _trigger_event
    self.document.callbacks.notify_event(cast(Model, self), event, invoke)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 247, in notify_event
    invoke_with_curdoc(doc, callback_invoker)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc
    return f()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 115, in invoke
    cast(EventCallbackWithEvent, callback)(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 398, in _comm_event
    self._process_bokeh_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 345, in _process_bokeh_event
    self._process_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/widgets/button.py", line 176, in _process_event
    self.clicks += 1
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 367, in _f
    instance_param.__set__(obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f
    return f(self, obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/__init__.py", line 625, in __set__
    super(Dynamic,self).__set__(obj,val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f
    return f(self, obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1248, in __set__
    obj.param._call_watcher(watcher, event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2039, in _call_watcher
    self_._execute_watcher(watcher, (event,))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/param.py", line 479, in action
    value(self.object)
  File "/home/jovyan/climakitae/warming_levels.py", line 268, in <lambda>
    lambda x: x.param.trigger("reload_data"), label="Reload Data", doc="Reload data"
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1989, in trigger
    self_.set_param(**dict(params, **triggers))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1925, in set_param
    return self_.update(kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1898, in update
    self_._batch_call_watchers()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2059, in _batch_call_watchers
    self_._execute_watcher(watcher, events)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 669, in caller
    return function()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 407, in _depends
    return func(*args, **kw)
  File "/home/jovyan/climakitae/warming_levels.py", line 306, in _update_postage_data
    self.postage_data = _get_postage_data(
  File "/home/jovyan/climakitae/warming_levels.py", line 38, in _get_postage_data
    data = _read_catalog_from_select(selections=selections, cat=cat)
  File "/home/jovyan/climakitae/data_loaders.py", line 585, in _read_catalog_from_select
    da = _get_data_one_var(selections, cat)
  File "/home/jovyan/climakitae/data_loaders.py", line 431, in _get_data_one_var
    dset = dset.rename({"lon": "x", "lat": "y"})
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3592, in rename
    variables, coord_names, dims, indexes = self._rename_all(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3548, in _rename_all
    variables, coord_names = self._rename_vars(name_dict, dims_dict)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3508, in _rename_vars
    raise ValueError(f"the new name {name!r} conflicts")
**ValueError: the new name 'x' conflicts**`

**Comment by bethem** [comment]
> in app.explore.warming(), when selecting Santa Cruz County @ 45 km (emphasis mine, since there are two main errors shown):
> 
> `Traceback (most recent call last): File "/home/jovyan/climakitae/data_loaders.py", line 425, in _get_data_one_var dset = dset.rio.clip(geometries=ds_region, crs=4326, drop=True) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_dataset.py", line 380, in clip self._obj[var] File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_array.py", line 941, in clip raise NoDataInBounds( **rioxarray.exceptions.NoDataInBounds: No data found in bounds. Data variable: t2**
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last): File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pyviz_comms/**init**.py", line 346, in _handle_msg self._on_msg(msg) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/viewable.py", line 291, in _on_msg patch.apply_to_document(doc, comm.id if comm else None) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in apply_to_document invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter)) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc return f() File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter)) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/document.py", line 391, in apply_json_patch DocumentPatchedEvent.handle_json(self, event_json, references, setter) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 259, in handle_json handler(doc, event_json, references, setter) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 300, in _handle_json cb(event_json["msg_data"]) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 355, in trigger_json_event model._trigger_event(event) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 119, in _trigger_event self.document.callbacks.notify_event(cast(Model, self), event, invoke) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 247, in notify_event invoke_with_curdoc(doc, callback_invoker) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc return f() File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 115, in invoke cast(EventCallbackWithEvent, callback)(event) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 398, in _comm_event self._process_bokeh_event(event) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 345, in _process_bokeh_event self._process_event(event) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/widgets/button.py", line 176, in _process_event self.clicks += 1 File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 367, in _f instance_param.**set**(obj, val) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f return f(self, obj, val) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/**init**.py", line 625, in **set** super(Dynamic,self).**set**(obj,val) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f return f(self, obj, val) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1248, in **set** obj.param._call_watcher(watcher, event) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2039, in _call_watcher self_._execute_watcher(watcher, (event,)) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher watcher.fn(*args, **kwargs) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/param.py", line 479, in action value(self.object) File "/home/jovyan/climakitae/warming_levels.py", line 268, in lambda x: x.param.trigger("reload_data"), label="Reload Data", doc="Reload data" File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1989, in trigger self_.set_param(**dict(params, **triggers)) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1925, in set_param return self_.update(kwargs) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1898, in update self_._batch_call_watchers() File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2059, in _batch_call_watchers self_._execute_watcher(watcher, events) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher watcher.fn(*args, **kwargs) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 669, in caller return function() File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 407, in _depends return func(*args, **kw) File "/home/jovyan/climakitae/warming_levels.py", line 306, in _update_postage_data self.postage_data = _get_postage_data( File "/home/jovyan/climakitae/warming_levels.py", line 38, in _get_postage_data data = _read_catalog_from_select(selections=selections, cat=cat) File "/home/jovyan/climakitae/data_loaders.py", line 585, in _read_catalog_from_select da = _get_data_one_var(selections, cat) File "/home/jovyan/climakitae/data_loaders.py", line 431, in _get_data_one_var dset = dset.rename({"lon": "x", "lat": "y"}) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3592, in rename variables, coord_names, dims, indexes = self._rename_all( File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3548, in _rename_all variables, coord_names = self._rename_vars(name_dict, dims_dict) File "/srv/conda/envs/notebook/lib/python3.9/site-packages/xarray/core/dataset.py", line 3508, in _rename_vars raise ValueError(f"the new name {name!r} conflicts") **ValueError: the new name 'x' conflicts**`

I'm guessing that the county doesn't fit into a 45 km grid cell. I think there's an option when clipping via a shapefile `all_touched=True` which clips even when the shapefile doesn't overlap with a grid center. Could be worth throwing that in for small counties, and maybe raising a warning to users.

**Review by bethem — CHANGES_REQUESTED** [review]
details in my comment

**Review by bethem — APPROVED** [review]
cool beans, worked just fine! I also tested Alameda, which did select a single grid point successfully before, but now includes more. this has the benefit of not giving us a scatter plot (it's a map!), but we might want to check with the others if `all_touched=True` is generally acceptable. note that it will impact area averages, so it's worth thinking about.

**Comment by elehmer** [comment]
I am still getting an error in `explore_warming` when selecting Santa Cruz County and 45km:

Traceback (most recent call last):
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pyviz_comms/__init__.py", line 346, in _handle_msg
    self._on_msg(msg)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/viewable.py", line 291, in _on_msg
    patch.apply_to_document(doc, comm.id if comm else None)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in apply_to_document
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc
    return f()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/protocol/messages/patch_doc.py", line 115, in <lambda>
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.content, setter))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/document.py", line 391, in apply_json_patch
    DocumentPatchedEvent.handle_json(self, event_json, references, setter)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 259, in handle_json
    handler(doc, event_json, references, setter)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/events.py", line 300, in _handle_json
    cb(event_json["msg_data"])
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 355, in trigger_json_event
    model._trigger_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 119, in _trigger_event
    self.document.callbacks.notify_event(cast(Model, self), event, invoke)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 247, in notify_event
    invoke_with_curdoc(doc, callback_invoker)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/document/callbacks.py", line 408, in invoke_with_curdoc
    return f()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/bokeh/util/callback_manager.py", line 115, in invoke
    cast(EventCallbackWithEvent, callback)(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 398, in _comm_event
    self._process_bokeh_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/reactive.py", line 345, in _process_bokeh_event
    self._process_event(event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/widgets/button.py", line 176, in _process_event
    self.clicks += 1
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 367, in _f
    instance_param.__set__(obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f
    return f(self, obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/__init__.py", line 625, in __set__
    super(Dynamic,self).__set__(obj,val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 369, in _f
    return f(self, obj, val)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1248, in __set__
    obj.param._call_watcher(watcher, event)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2039, in _call_watcher
    self_._execute_watcher(watcher, (event,))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/panel/param.py", line 479, in action
    value(self.object)
  File "/home/jovyan/src/climakitae/climakitae/warming_levels.py", line 256, in <lambda>
    lambda x: x.param.trigger("reload_data"), label="Reload Data", doc="Reload data"
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1989, in trigger
    self_.set_param(**dict(params, **triggers))
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1925, in set_param
    return self_.update(kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 1898, in update
    self_._batch_call_watchers()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2059, in _batch_call_watchers
    self_._execute_watcher(watcher, events)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 2021, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 669, in caller
    return function()
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/param/parameterized.py", line 407, in _depends
    return func(*args, **kw)
  File "/home/jovyan/src/climakitae/climakitae/warming_levels.py", line 294, in _update_postage_data
    self.postage_data = _get_postage_data(
  File "/home/jovyan/src/climakitae/climakitae/warming_levels.py", line 39, in _get_postage_data
    data = _read_catalog_from_select(selections=selections, cat=cat)
  File "/home/jovyan/src/climakitae/climakitae/data_loaders.py", line 638, in _read_catalog_from_select
    da = _get_data_one_var(selections, cat)
  File "/home/jovyan/src/climakitae/climakitae/data_loaders.py", line 432, in _get_data_one_var
    dset = dset.rio.clip(
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_dataset.py", line 380, in clip
    self._obj[var]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/rioxarray/raster_array.py", line 941, in clip
    raise NoDataInBounds(
rioxarray.exceptions.NoDataInBounds: No data found in bounds. Data variable: t2

**Pull request merged**